cloud-optimized-geotiffs/writing-cogs-in-python.ipynb

```diff
@@ -0,0 +1,525 @@
+{
```

Comment:
<p>predictor=2 is not explained</p><p>blocksize; so that's the internal tile size, what about overview block size? also not explained</p><p>nodata is rarely set to 0, since 0 often has meaning, Int would be largest negative value or -9999, float has a nodata option, Byte as discussed is a mask band. Maybe this is a bad example to use - looking at <a href="https://lpdaac.usgs.gov/documents/1318/NASADEM_User_Guide_V12.pdf" target="_blank">https://lpdaac.usgs.gov/documents/1318/NASADEM_User_Guide_V12.pdf</a> there's a 2nd file the NUM file, and 0 in that band corresponds to water aka NoData if you wanted to mask the water out.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/cloudnativegeo/cloud-optimized-geospatial-formats-guide/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Reply:
<p>Added following text explaining overviews and <code>NoData</code> values-</p><p> </p>

<pre class="ql-syntax" spellcheck="false">Here, we will see how to -

Generate COGs with overviews in Python.

Overviews are decimated views created for visualization. By default,&nbsp;rio-cogeo&nbsp;calculates optimal overview levels based on dataset size and internal tile size, since overviews should not be smaller than internal tile size.

Set&nbsp;NoData&nbsp;values while generating COGs.

The&nbsp;NoData&nbsp;value can be set based on the datatype used for storing the data. For&nbsp;int&nbsp;type, the largest negative value or&nbsp;-9999&nbsp;is used. Similarly,&nbsp;float&nbsp;type has a&nbsp;NoData&nbsp;option and&nbsp;Byte&nbsp;has a mask band to define the&nbsp;NoData&nbsp;value. 
</pre>

Reply:
<p>Sure but in this case the HGT has:</p>

<pre class="ql-syntax" spellcheck="false">Band 1 Block=3601x1 Type=Int16, ColorInterp=Undefined
&nbsp; NoData Value=-32768
&nbsp; Unit Type: m
</pre><p>so we should use that value. you should be able to get that from the <code>src</code> metadata when reading with rasterio and keep it for writing out. </p>

Reply:
Thank you for pointing this out. I've added following description along with a line showing how to print metadata and retrieve nodata value - 

"
Typically, while creating a COG from a Non-COG TIFF, the nodata value can be set to the value defined in the source GeoTIFF metadata (in this case -32768.0). Otherwise, the nodata value can be set based on the datatype used for storing the data. For int type, the largest negative value or -9999 is used. Similarly, float type has a nodata option and Byte has a mask band to define the nodata value.
"

Also, updated the nodata value in the code illustration.